compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -118,55 +239,66 @@ static std::ostream& indentStream(std::ostream& os, size_t num) {
   return os;
 }
 
-// Remove the leading+trailing // or /* (*/)
+// Remove the leading+trailing commentStyle (/*+, *+/)
 // Dedent by the least amount of leading whitespace
-// Return is a list of strings which have no newline chars
-static std::vector<std::string> prettifyComment(const std::string& s) {
-  std::string ret = s;
-
-  if (ret.substr(0, 2) == "//") {
-    ret.erase(0, 2);
-  } else if (ret.substr(0, 2) == "/*") {
+// Return is a list of strings which may have newline chars
+static std::vector<std::string> prettifyComment(const std::string& comment,
+                                                const std::string& commentStyle) {
+  std::string ret = comment;
+  std::string commentEnd = commentStyle;
+  reverse(commentEnd.begin(), commentEnd.end());
+  const int styleLen = commentStyle.size();
+  if (ret.substr(0, styleLen) == commentStyle) {
     size_t l = ret.length();
-    assert(ret.substr(l - 2, 2) == "*/" && "comment didn't end with */");
-    ret.erase(l - 2, 2);
-    ret.erase(0, 2);
+    if (ret.substr(l - styleLen, styleLen) != commentEnd)
+      // TODO: this is a broken comment at this point, no?
+      return {};
+    ret.erase(l - styleLen, styleLen);
+    ret.erase(0, styleLen);
   } else {
-    assert(false && "comment didn't start with // or /*");
+    return {};
   }
 
-  ret = strip(ret, "^\n+|\n+$");
-
   auto lines = splitLines(ret);
   if (lines.empty()) return lines;
 
   size_t toTrim = std::numeric_limits<size_t>::max();
-  for (const std::string& line : lines) {
-    if (line.empty()) continue;
-    toTrim = std::min(toTrim, countLeadingSpaces(line));
+  // exclude the first line from the minimum
```

Comment:
yes, we strip the spaces from the first line to treat it as the initial position, but don't count the leading spaces of the first line because it will already have been modified by removing the comment indicator.